src/bin/cross-util.rs

```diff
@@ -49,39 +49,56 @@ fn is_toolchain(toolchain: &str) -> cross::Result<String> {
 
 fn get_container_engine(
     engine: Option<&str>,
-    msg_info: MessageInfo,
+    docker_in_docker: bool,
+    msg_info: &mut MessageInfo,
 ) -> cross::Result<docker::Engine> {
     let engine = if let Some(ce) = engine {
         which::which(ce)?
     } else {
         docker::get_container_engine()?
     };
-    docker::Engine::from_path(engine, None, msg_info)
+    let in_docker = match docker_in_docker {
+        true => Some(true),
+        false => None,
+    };
+    docker::Engine::from_path(engine, in_docker, None, msg_info)
+}
+
+macro_rules! get_engine {
+    ($args:ident, $docker_in_docker:expr, $msg_info: ident) => {{
+        get_container_engine($args.engine(), $docker_in_docker, &mut $msg_info)
```

Comment:
These macros just simplify a lot of redundant code that existed earlier, while being relatively transparent.